src/Listeners/ProcessUpdate.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Telegram\Bot\Listeners;
+
+use Telegram\Bot\Events\UpdateReceived;
+use Telegram\Bot\Exceptions\TelegramSDKException;
+use Throwable;
+
+class ProcessUpdate
+{
+
+    /**
+     * @param UpdateReceived $event
+     *
+     * @throws TelegramSDKException
+     */
+    public function handle(UpdateReceived $event)
+    {
+        //Determine the type of update and fire an event.
+        $eventName = $event->getUpdate()->getEvent();
+
+        try {
+            $event->getBot()->getEventFactory()->dispatch(
+                $eventName,
+                new $eventName($event->getBot(), $event->getUpdate())
+            );
+        } catch (Throwable $e) {
+            //TODO
+            // What should happen if there was a new update type added to telegram but
+            // we haven't updated the SDK yet?
```

Comment:
Maybe we can dispatch a generic event here in that case. Let me test a few things. For now, let's leave this as is but just remove the comment.

Reply:
Don't need to anymore! It's much simpler.